linera-core/src/client/mod.rs

```diff
@@ -1541,40 +1535,77 @@ where
 
     /// Downloads and processes from the specified validator a confirmed block certificate that
     /// uses the given blob. If this succeeds, the blob will be in our storage.
-    async fn update_local_node_with_blob_from(
+    async fn update_local_node_with_blobs_from(
         &self,
-        blob_id: BlobId,
+        blob_ids: Vec<BlobId>,
         remote_node: &RemoteNode<P::Node>,
     ) -> Result<(), ChainClientError> {
-        let certificate = remote_node.download_certificate_for_blob(blob_id).await?;
-        // This will download all ancestors of the certificate and process all of them locally.
-        self.receive_certificate(certificate).await?;
+        try_join_all(blob_ids.into_iter().map(|blob_id| async move {
+            let certificate = remote_node.download_certificate_for_blob(blob_id).await?;
+            // This will download all ancestors of the certificate and process all of them locally.
+            self.receive_certificate(certificate).await
+        }))
+        .await?;
```

Comment:
Would it be better to have a limit for how many tasks we create here? Maybe using `buffer_unordered`? (Something like I tried in https://github.com/linera-io/linera-protocol/pull/2588.)

Reply:
We could 🤔 But won't the tokio runtime's thread pool handle if the validator is getting too many requests? As in it'll queue the requests and wait until a thread is available to execute it?
If that works well, then we can always have the Validator at "full capacity". If we limit here, we could be slowing things down even if the validator has idle threads, if we guess a number that is too small 🤔
Happy to do it, but wanted to see what your thoughts are on this first, maybe I'm missing something